libcloudforensics/providers/aws/internal/ec2.py

```diff
@@ -137,3 +142,368 @@ def AttachVolume(self,
+      cpu_cores (int): Number of CPU cores for the analysis VM.
+      boot_volume_type (str): Optional. The volume type for the boot volume
+          of the VM. Can be one of 'standard'|'io1'|'gp2'|'sc1'|'st1'. The
+          default is 'gp2'.
+      packages (List[str]): Optional. List of packages to install in the VM.
+      ssh_key_name (str): Optional. A SSH key pair name linked to the AWS
+          account to associate with the VM. If none provided, the VM can only
+          be accessed through in-browser SSH from the AWS management console
+          with the EC2 client connection package (ec2-instance-connect). Note
+          that if this package fails to install on the target VM, then the VM
+          will not be accessible. It is therefore recommended to fill in this
+          parameter.
+      tags (Dict[str, str]): Optional. A dictionary of tags to add to the
+          instance, for example {'TicketID': 'xxx'}. An entry for the instance
+          name is added by default.
+
+    Returns:
+      Tuple[AWSInstance, bool]: A tuple with an AWSInstance object and a
+          boolean indicating if the virtual machine was created (True) or
+          reused (False).
+
+    Raises:
+      RuntimeError: If the virtual machine cannot be found or created.
+    """
+
+    # Re-use instance if it already exists, or create a new one.
+    try:
+      instances = self.GetInstancesByName(vm_name)
+      if instances:
+        created = False
+        return instances[0], created
+    except RuntimeError:
+      pass
+
+    instance_type = common.GetInstanceTypeByCPU(cpu_cores)
+    startup_script = utils.ReadStartupScript()
+    if packages:
+      startup_script = startup_script.replace('${packages[@]}', ' '.join(
+          packages))
+
+    # Install ec2-instance-connect to allow SSH connections from the browser.
+    startup_script = startup_script.replace(
+        '(exit ${exit_code})',
+        'apt -y install ec2-instance-connect && (exit ${exit_code})')
+
+    if not tags:
+      tags = {}
+    tags['Name'] = vm_name
+
+    client = self.aws_account.ClientApi(common.EC2_SERVICE)
+    vm_args = {
+        'BlockDeviceMappings':
+            [self._GetBootVolumeConfigByAmi(
+                ami, boot_volume_size, boot_volume_type)],
+        'ImageId': ami,
+        'MinCount': 1,
+        'MaxCount': 1,
+        'InstanceType': instance_type,
+        'TagSpecifications': [common.CreateTags(common.INSTANCE, tags)],
+        'UserData': startup_script,
+        'Placement': {
+            'AvailabilityZone': self.aws_account.default_availability_zone}
+    }
+    if ssh_key_name:
+      vm_args['KeyName'] = ssh_key_name
+    # Create the instance in AWS
+    try:
+      instance = client.run_instances(**vm_args)
+      # If the call to run_instances was successful, then the API response
+      # contains the instance ID for the new instance.
+      instance_id = instance['Instances'][0]['InstanceId']
```

Comment:
Could also throw IndexError/KeyError?

Reply:
If the call is successful it's expected to have the correct structure. If the call isn't we rely on the `boto3` library to throw an exception for whatever reason it couldn't do its job